### ISSUE TITLE
Password must contain numbers not in first 100k digits of Pi

### ISSUE THREAD

**Issue by BenBE** [troyhunt/password-purgatory-api#49]
Include a check that each number inside the password must not be contained as a substring of the first 100k digits of [pi](http://www.geom.uiuc.edu/~huberty/math5337/groupe/digits.html). This enforces, that no numbers with 1, 2, 3, or 4 digits may be used and excludes ~63% of all 5-digit numbers (63279). For 6-digit numbers ~9.5% (95060) are excluded by this.

For extra fun, repeat this with the first 100k digits of [e](https://en.wikipedia.org/wiki/E_(mathematical_constant)) …

**Comment by MichaelNMaggs** [comment]
I don't think that will encourage people to keep going, which is what we want. Rather than guessing at an answer, most will just give up

**Comment by BenBE** [comment]
You can calibrate the complexity by varying the number of used digits. Tenth of the digits and you allow for shorter runs of digits to be used. With 10k digits used you could use roughly 1/3 of all 4 digit numbers; with just 1k of digits you are below 10% of 4-digit numbers not acceptable, which should be easy enough to defeat to not encourage giving up.

**Comment by MichaelNMaggs** [comment]
Agreed that it would be easy enough to defeat, but the vast majority of people will have no idea how to do that, and even if they did the message is so obviously trolling that they'll back out. Messages should be just plausible enough to encourge victims to think they are real, and to continue
